application/core/LSHttpRequest.php

```diff
@@ -286,4 +285,21 @@ public function isRestRequest()
         ) === 1;
         return $restPath || $restRoute;
     }
+
+    /**
+     * @inheritdoc
+     * Check host with config['allowedHost'] if it set
+     */
+    public function getHostInfo($schema = '')
+    {
+        $hostInfo = parent::getHostInfo($schema);
+        if (!empty(App()->getConfig('allowedHosts')) && is_array(App()->getConfig('allowedHosts'))) {
```

Comment:
From here and until the end of the function, wouldn't it be better to make it a function of its own?
CheckAllowedHost() or IsAllowedHost($host)

Then, the function will be easier to automatically test and enrich in the future.

Reply:
If you set a bad host : you get a 400 error on all admin page : easy to test

Here it's a 5 lines of code : really need a specific function ?

Reply:
> If you set a bad host : you get a 400 error on all admin page : easy to test

I mean automatic testing.

> Here it's a 5 lines of code : really need a specific function ?

Well, it wouldn't be the worst having a 5 lines function either.
If the rules get more complex, and it is not on a function of its own, automatic testing would get harder.

Reply:
> I mean automatic testing.

Yes, it's easy to add it :) : just call admin page and check if its a 400 header.

> Well, it wouldn't be the worst having a 5 lines function either.

Separated